src/txn/txn.c

```diff
@@ -441,20 +441,61 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 	if (cval.len > 0) {
 #ifdef HAVE_TIMESTAMPS
 		wt_timestamp_t ts;
+		WT_TXN_GLOBAL *txn_global;
+		char timestamp_buf[2 * WT_TIMESTAMP_SIZE + 1];
+		bool round_to_oldest;
 
+		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-		WT_RET(__wt_timestamp_validate(session,
-		    &ts, &cval, true, false, false));
-		__wt_timestamp_set(&txn->read_timestamp, &ts);
+		WT_RET(__wt_config_gets_def(session, cfg, "round_to_oldest", 0,
+		    &cval));
+		round_to_oldest = false;
+		__wt_readlock(session, &txn_global->rwlock);
+		if (__wt_timestamp_cmp(&ts, &txn_global->oldest_timestamp) < 0)
+		{
+			/*
+			 * To circumvent error: unused-result
+			 */
+			(void)(__wt_timestamp_to_hex_string(session,
```

Comment:
You should do WT_RET(__wt_timestamp_to_hex_string(..));

That will get rid of compiler complaining about unused return. Also if this function fails the error will be handled gracefully.